includes/islandora_multi_batch.inc

```diff
@@ -481,52 +481,102 @@ class IslandoraMultiBatchObject extends IslandoraBatchObject {
+      // Use object_info to create some datastreams.
+      $this->label = !empty($this->getTitle()) ? $this->getTitle() : "New Object";
+      $errors = array();
+      $files = array();
+      $datastreams = array();
+      $datastreams = $this->getDatastreams($errors, $files);
+      if (!empty($errors)) {
+          $setid = $this->getBatchId();
+          foreach ($errors as $error) {
+              $parms = array('@setid' => $setid, '@pid' => $this->id, '@log' => $error );
+              $msg = 'There were some issues on Batch set id @setid when generating datastreams for PID @pid: @log';
+              watchdog('islandora_multi_importer', $msg, $parms, WATCHDOG_NOTICE);
+          }
+      }
+      if (empty($datastreams)) {
+          $this->currentState = ISLANDORA_BATCH_STATE__ERROR;
+          return array('message' => 
+              t('Serious datastream issues on @pid have forced us to not ingest it.',
+          array('@pid' => $this->id)), 'state' => ISLANDORA_BATCH_STATE__ERROR);
       }
-    }
-    if (empty($datastreams)) {
-      $this->currentState = ISLANDORA_BATCH_STATE__ERROR;
-      return array('message' => 
-        t('Serious datastream issues on @pid have forced us to not ingest it.',
-        array('@pid' => $this->id)), 'state' => ISLANDORA_BATCH_STATE__ERROR);
-    }
     
-    foreach ($files as $file) {
-      $file->status &= ~FILE_STATUS_PERMANENT; // Lets make all files temporary but trust file_usage to keep them around
-      file_save($file);
-      file_usage_add($file, 'islandora_batch', 'islandora_batch_object', $this->getBatchId());
-    }
-    // We need this to check if we have mods
-    $dowehaveMODS = false;
+      foreach ($files as $file) {
+          $file->status &= ~FILE_STATUS_PERMANENT; // Lets make all files temporary but trust file_usage to keep them around
+          file_save($file);
+          file_usage_add($file, 'islandora_batch', 'islandora_batch_object', $this->getBatchId());
+      }
+      // We need this to check if we have mods
+      $dowehaveMODS = false;
+      $dowehaveDC = false;
+      $dcxml = FALSE;
+      foreach ($datastreams as $ds) {
+          $ds_uri = isset($ds['file']) ?
+              $ds['file']->uri :
+          $ds['datastream_file'];
+
+          $datastream = $this->constructDatastream($ds['dsid'], $ds['control_group']);
+          $datastream->label = $ds['label'];
+          $datastream->mimetype = $ds['mimetype'];
     
-    foreach ($datastreams as $ds) {
-      $ds_uri = isset($ds['file']) ?
-        $ds['file']->uri :
-        $ds['datastream_file'];
-
-      $datastream = $this->constructDatastream($ds['dsid'], $ds['control_group']);
-      $datastream->label = $ds['label'];
-      $datastream->mimetype = $ds['mimetype'];
+          $datastream->setContentFromFile($ds_uri);
+          $this->ingestDatastream($datastream);
+          $dowehaveMODS = $ds['dsid'] == "MODS" ? true : false; 
+          $dowehaveDC = $ds['dsid'] == "DC" ? true : false; 
```

Comment:
There's an error in the logic here for tracking the presence of MODS and DC datastreams. Will be false if the DS in question is not the last one.
Should be something like:
          if($ds['dsid'] == "MODS") $dowehaveMODS = TRUE;
          if($ds['dsid'] == "DC") $dowehaveDC = TRUE;

Reply:
Totally, what a mess. fixing now

Reply:
So what about $dowehaveMODS = $ds['dsid'] == "MODS" ? true : $dowehaveMODS;  ?
I think that was my original idea.

Reply:
So found another error after fixing this one ( on the collection policy!), fixed too. The worst way of saying goodbye to this year, brain is mashed potatoes. New logic assigns the previous value, so if set to true at any point, true will stick. Can you check my brain is doing the right logic there? Thanks

Reply:
Works for me. Seems slightly redundant to be assigning on each time through the loop when we know that it's starting as FALSE, and only need to change when TRUE. But mostly I just think it's a matter of coder preference.